packages/react-hooks/src/useSetInterval.ts

```diff
@@ -0,0 +1,30 @@
+import { useConst } from './useConst';
+import { useEffect } from 'react';
+
+/**
+ * Returns a self disposing setTimeout function.
+ */
+export const useSetInterval = () => {
+  const intervalIds = useConst<number[]>([]);
+
+  useEffect(
+    () => () => {
+      for (const id of intervalIds) {
+        clearInterval(id);
+      }
+      intervalIds.splice(0);
+    },
+    [],
+  );
+
+  return (func: () => void, duration: number) => {
```

Comment:
you can make the typing better:
```
type setIntervalType = typeof window.setInterval;

....

return (...args: Parameters<setIntervalType>): ReturnType<setIntervalType> => {
  ...
}
```

Reply:
Do you need a `...` for passing `rest` to the real `setInterval`?